radix-engine-stores/src/interface.rs

```diff
@@ -191,11 +195,11 @@ pub trait SubstateDatabase {
         &self,
         node_id: &NodeId,
         module_id: ModuleId,
-        substate_key: SubstateKey,
+        substate_key: &SubstateKey,
     ) -> Option<D> {
         self.get_substate(
             &M::map_to_db_index(node_id, module_id),
```

Comment:
I've suggested to Jakub that we rename some of these and refactor this mapper to not be generic (as the engine _relies_ on the concrete implementation of the mapper for ordering of the index)

Reply:
The engine does rely on this, but the engine relies on Track to uphold this invariant NOT the database. I'd still like the database layer to be abstracted from this.